react-form-controls/my-app/src/RegistrationFormControlled.tsx

```diff
@@ -0,0 +1,41 @@
+import { useState } from 'react';
+
+export default function RegistrationFormUncontrolled() {
+  const [userName, setUserName] = useState('');
+  const [password, setPassword] = useState('');
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    console.log({ userName, password });
+  }
+
+  return (
+    <form
+      onSubmit={handleSubmit}
+      style={{
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'flex-end',
+      }}>
+      <label htmlFor="username">
```

Comment:
You don't need `htmlFor` if you wrap the label around the input, as you have done. Also, `htmlFor` targets a component with the specified `id`, but your `input` does not have an `id`!